javascript.md

```diff
@@ -20,10 +20,30 @@ This one is already mentioned in [Steve Kwan's best practices][steve-kwan-best-p
 - Get peer reviews of your code: Make 'em short; do them early; small reviews get much faster feedback.
 - New files need to be added to the build profile.
 - When native javascript APIs exist, use these instead of those provided by a framework. i.e. Native `array.forEach` vs `dojo.base.array.forEach`.
-- Similarly, whenever possible, use native HTML5 widgets over framework widgets such as dojo's `dijit`. Only use other widgets when absolutely necessary.
+- Similarly, whenever possible, use native HTML5 widgets over framework widgets such as dojo's `dijit`. Only use other widgets when absolutely necessary and preferably instantiate those widgets programmatically.
 - Use [JSDoc3][jsdoc3] to document your code.
-- Write *TESTS*! Especially for your models
-- Define your CSS at component level instead of page (and put each into it's own component file)
+- Write *TESTS*! Especially for your models.
+- Define your CSS at component level instead of page (and put each into it's own component file).
+- Use compass mixins instead of multiple rules with browser prefixes.
+- Don't use `parseOnLoad`, parse small page sections instead.
+- Minimize DOM manipulation, use an in memory DOM if necessary to minimize the performance issues related to DOM layout recalculation.
+- Use `lang.clone` just when it is strictly necessary, it can cause performance issues.
+- Use the following [jshint file][jshint-file]
+- Minimize the number of delete statements in order to prevent performance issues:
+
+  ``` javascript
+  // Remove hidden classes...
+  delete object.a;
+
+  // Use null set instead so the hidden classes are still valid.
+  object.a = null;
+  ```
+
+  Some links to understand the rule above:
+
+  1. [Google v8 optimizations][google-v8-optimizations].
+  2. Addy Osmani - Memory Management [keynote][memory-management-keynote] and [video][memory-management-video]
+
```

Comment:
To keep the guidelines to a reasonable size, I'd suggest to link part of the bullet point, instead of explicitly add text just for the links.